jobs/tests.py

```diff
@@ -1,3 +1,82 @@
 from django.test import TestCase
+from django.utils import timezone
 
-# Create your tests here.
+from model_mommy import mommy
+
+from jobs.models import Job
+
+
+class JobModelTests(TestCase):
+    """Tests for custom methods of Job model."""
+
+    """Setting up models with mommy."""
+    def setUp(self):
+        self.job = mommy.make(Job, ready_to_publish=True)
+        self.job1 = mommy.make(Job, ready_to_publish=False)
+
+    """Plain publish method test without any twists."""
+    def test_publish_succesfully(self):
+        self.assertFalse(self.job.published_date)
+        self.job.publish()
+        self.assertTrue(self.job.published_date, "Job has no published date.")
+
+    """Publish doesn't work when ready_to_publish field is set to False."""
+    def test_publish_unsuccesfully(self):
+        self.assertFalse(self.job1.published_date)
+        self.job1.publish()
+        self.assertFalse(self.job1.published_date, "Job has published date.")
+
+    """Publish once, than change ready_to_publish field value and publish once more."""
+    def test_publish_more_than_once_and_change_values(self):
+        self.assertFalse(self.job.published_date)
+        self.job.publish()
+        self.job.ready_to_publish == False
+        # Right now if we change ready_to_publish to False, published_date stays as it was and doesn't become None.
```

Comment:
I am not sure if we want this to happen.. if we republish than date changes. If old date stays until republish we know that it was already published and taken down, it may be good knowledge to have, but also it may be not really useful.